scripts/deploy-token.ts

```diff
@@ -18,16 +18,10 @@ async function main() {
   const STMXContract = await ethers.getContractFactory('STMX', owner)
 
   // 18 decimal places, convert to full units
-  const initialSupply = BigNumber.from(values.INITIAL_SUPPLY).mul(BigNumber.from(10).pow(18))
-  const args = [values.NAME, values.SYMBOL, initialSupply, owner.address]
-  const token = await upgrades.deployProxy(STMXContract, args) as STMX
-
-  console.info(`Check tx here: ${getEtherscanUri('tx', token.deployTransaction.hash)}`)
-
-  await token.deployed()
```

Comment:
How do you deploy if this is removed?

Reply:
.deployed() is only used to wait for the deployment. The contract is actually deployed by calling the method .deploy(), which is already there. 

Also, this .deployed() method is not available in the new versions that why I removed it. Now I have added the alternate which is .waitForDeployment()